mongodb/trigger/eventlistener/metadata.go

```diff
@@ -0,0 +1,96 @@
+package eventlistener
+
+import (
+	"github.com/project-flogo/core/data/coerce"
+)
+
+// Settings structure
+type Settings struct {
+	Connection string `md:"mongodbConnection,required"`
+}
+
+// HandlerSettings structure
+type HandlerSettings struct {
+	Collection   string `md:"collection"`
+	ListenInsert bool   `md:"listenInsert,required"`
+	ListenUpdate bool   `md:"listenUpdate,required"`
+	ListenRemove bool   `md:"listenRemove,required"`
+}
+
+// Output structure
+type Output struct {
+	Output map[string]interface{} `md:"Output"`
+}
+
+// FromMap method for HandlerSettings
+func (i *HandlerSettings) FromMap(values map[string]interface{}) error {
+	var err error
+
+	i.Collection, err = coerce.ToString(values["collection"])
+	if err != nil {
+		return err
+	}
+	i.ListenInsert, err = coerce.ToBool(values["listenInsert"])
+	if err != nil {
+		return err
+	}
+	i.ListenUpdate, err = coerce.ToBool(values["listenUpdate"])
+	if err != nil {
+		return err
+	}
+	i.ListenRemove, err = coerce.ToBool(values["listenRemove"])
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//ToMap method for HandlerSettings
+func (i *HandlerSettings) ToMap() map[string]interface{} {
+	return map[string]interface{}{
+		"collection":   i.Collection,
+		"listenInsert": i.ListenInsert,
+		"listenUpdate": i.ListenUpdate,
+		"listenRemove": i.ListenRemove,
+	}
+}
+
+//FromMap method for Settings
+func (i *Settings) FromMap(values map[string]interface{}) error {
```

Comment:
Any particular reason for "FromMap" and "ToMap" functions for settings ?

Reply:
A developer from our team who worked on this file insisted that we need to have this FromMap and ToMap for settings as well else the build or running app didn't work. Let me re-check by removing these functions. If building and running the app works even without, I will remove these 2 functions.

Reply:
Okay!  This is not needed.